### PR TITLE
Add a check for crio service before starting node e2e tests

### LINKED ISSUE #94759: crio node e2e test fails
**Which jobs are failing**:
/test pull-kubernetes-node-crio1-18-e2e

**Which test(s) are failing**:
pull-kubernetes-node-crio1-18-e2e

**Reason for failure**:
Just before node e2e tests are started, we verify that the runtime (containerd or docker) is properly running or not on the target system. This check doesn't include crio, so the targeted node is not reported ready if it's using crio as the runtime. 

```bash
I0909 10:32:31.445] 
I0909 10:32:31.446] >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
I0909 10:32:31.447] >                              START TEST                                >
I0909 10:32:31.447] >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
I0909 10:32:31.447] Start Test Suite on Host 
I0909 10:32:31.447] 
I0909 10:32:31.447] Failure Finished Test Suite on Host 
I0909 10:32:31.448] unable to create gce instance with running docker daemon for image fedora-coreos-32-20200824-3-0-gcp-x86-64.  instance tmp-node-e2e-7eb5b1c5-fedora-coreos-32-20200824-3-0-gcp-x86-64 not running docker/containerd daemon - Command failed: prow@104.199.120.168: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
I0909 10:32:31.448] 
I0909 10:32:31.448] <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
I0909 10:32:31.448] <                              FINISH TEST                               <
I0909 10:32:31.448] <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
I0909 10:32:31.448] 
```

### PR DESCRIPTION
**Pull request by harche**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #94759 

**Special notes for your reviewer**:
`None`

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: test/e2e_node/remote/ssh.go [hunk 1]
```diff
@@ -70,7 +70,8 @@ func GetHostnameOrIP(hostname string) string {
 		host = ip
 	}
 
-	if *sshUser == "" {
+	// Do no ignore the KUBE_SSH_USER if it's set explicitly
+	if os.Getenv("KUBE_SSH_USER") != "" {
```
Comment (oomichi) [inline test/e2e_node/remote/ssh.go:74]: I think the specified sshUser should be used instead of the env variable if specifying `--ssh-user` option.

**Review by oomichi — COMMENTED** [review]
/cc @oomichi

File: test/e2e_node/remote/ssh.go (hunk 1 above)
Comment (harche) [reply test/e2e_node/remote/ssh.go:74]: @oomichi Thanks for your reply. For the Fedora CoreOS host, the only user available on host is `core`.  We have seen failures like, 

```
prow@104.199.120.168: Permission denied (publickey,gssapi-keyex,gssapi-with-mic)
```
The `prow` user doesn't exist on the Fedora CoreOS. In fact, `KUBE_SSH_USER` was [introduced](https://github.com/kubernetes/kubernetes/pull/94049/files) to handle Fedora CoreOS hosts (which will be used extensively for CRI-O testing as well as in future RuntimeClass tests)

This PR doesn't prevent the existing flow, it's just that for Fedora CoreOS we want to make sure we use the username that we know works. 

cc @dims

Comment (oomichi) [reply test/e2e_node/remote/ssh.go:74]: @harche Thanks for your response. I see your point.
I guess that comes from https://github.com/kubernetes/test-infra/blob/master/kubetest/e2e.go#L575 which gets the specified username on `--ssh-user` option from the env value `USER`.
I feel it is better to get the username from the env value `KUBE_SSH-USER` if existing. If not existing, get the username from the env value `USER` as current implementation.

I'd like to get help from @BenTheElder for the above idea.

**Comment by dims** [comment]
@oomichi let's unblock the CRI-O folks as we know this works. We can log a follow up issue to tweak if needed. How does that sound?

**Comment by oomichi** [comment]
> @oomichi let's unblock the CRI-O folks as we know this works. We can log a follow up issue to tweak if needed. How does that sound?

As you know, for example kubectl tries reading the specified kubeconfig at first. If not specified, the env value KUBECONFIG is read. Such behavior seems pretty straightforward and that is the same as the existing one.
And current PR changes the above behavior like makes the env value stronger than the specified option value.
I'd like to see opinions from other people.

File: test/e2e_node/remote/ssh.go (hunk 1 above)
Comment (BenTheElder) [reply test/e2e_node/remote/ssh.go:74]: I agree that if --ssh-user is set we should respect that.
It also seems like https://github.com/kubernetes/test-infra/blob/feccbf60454db394d955e00977a9e2829712f050/kubetest/e2e.go#L575 could handle KUBE_SSH_USER overridding USER

File: test/e2e_node/remote/ssh.go [hunk 2]
```diff
@@ -70,7 +70,8 @@ func GetHostnameOrIP(hostname string) string {
...
 		*sshUser = os.Getenv("KUBE_SSH_USER")
```
Comment (neolit123) [inline test/e2e_node/remote/ssh.go:75]: as discussed above the flag should take precedence over the environment variable.

please change the logic to something like:

```
var user string
if *sshUser != "" {
	user = *sshUser
	goto add_user
}
if os.Getenv("KUBE_SSH_USER") != "" {
	user = os.Getenv("KUBE_SSH_USER")
	goto add_user
}
return host
add_user:
	return fmt.Sprintf("%s@%s", user, host)
```

**Comment by harche** [comment]
@oomichi @BenTheElder @dims based on the feedback, I have created https://github.com/kubernetes/test-infra/pull/19273 to set the ssh username. 

The only change in this PR now is just to add the check to see if the remote node is running `crio.service` or not. 

Thanks everyone for taking time out for reviewing this PR.

File: test/e2e_node/remote/ssh.go (hunk 1 above)
Comment (harche) [reply test/e2e_node/remote/ssh.go]: https://github.com/kubernetes/test-infra/pull/19273  making `kubetest/e2e.go` handle KUBE_SSH_USER overridding USER. 

cc @oomichi @BenTheElder

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Comment by neolit123** [comment]
i've commented on the test-infra PR.
/priority important-longterm

**Comment by oomichi** [comment]
Thanks for your work @harche 

/lgtm

**Comment by harche** [comment]
> /approve
> /lgtm
> /milestone v1.20

Thanks @dims, it seem for some reason the `approve` label is not getting assigned.

**Comment by dims** [comment]
@harche i don't have approval rights for this directory :)

**Comment by harche** [comment]
> @harche i don't have approval rights for this directory :)

oh ok. @sjenning do you have the approval rights?

**Comment by neolit123** [comment]
Ideally someone from #sig-node (k8s slack)should approve.

**Pull request merged**